### ISSUE TITLE
Grid visibility: live training metrics, node SSH, per-node widgets, Foreman alerts

### ISSUE THREAD

**Issue by joelteply** [CambrianTech/continuum#682]
## Summary

Everything Claude does over SSH today needs to be visible and actionable through the grid UI. Training metrics, node health, process management, log tailing, quality gates — all surfaced through commands, events, and widgets. Plus Foreman autonomously monitoring and alerting.

### Monitoring
- `nvidia-smi` — GPU utilization, VRAM, temperature
- `tail -f /tmp/forge.log` — live training output
- `ps aux | grep python` — running processes
- `df -h` — disk space
- `python3 -c "import torch; ..."` — environment health

### Detecting Problems
- **Perplexity spiraling** (3.04 → 26.92 like today's forge)
- **Loss plateauing or diverging**
- **Process died** (zombie PIDs, OOM kills)
- **Disk full** (forge can't write outputs)
- **Wrong PyTorch/CUDA** (sm_120 incompatibility)
- **Model quality degradation** (bad samples mid-training)

### Fixing Problems
- Kill zombie processes
- Clear disk space
- Upgrade packages
- Restart failed jobs from checkpoint
- Adjust hyperparameters mid-run

### Per-Node Tab (Click a node in Grid view → opens as a tab)
Each grid node should be a navigable entity — click it in the Grid overview and it opens as its own tab with:

- **Terminal** — SSH/shell access right in the browser tab
- **Right panel widgets** specific to that node:
  - GPU stats (live)
  - Disk usage
  - Running processes with kill buttons
  - Environment health (Python, CUDA, packages)
  - Foreman chat (if assigned)
- **AI persona** can be brought into the node context to help diagnose

### Training Tab (Active training sessions)
When a forge or LoRA training is running, its own tab with:

- **Live metrics**: loss curve, perplexity, learning rate schedule, gradient norm
- **Samples**: code/text generated at checkpoints (shown inline)
- **Quality gates**: perplexity vs baseline, improvement %, pass/fail threshold
- **Stage progress**: which alloy stage is active, how many cycles done
- **Right panel widgets**:
  - Loss sparkline (zoomed)
  - Perplexity trend vs baseline
  - Hardware utilization during training
  - Alloy recipe summary
  - Estimated time remaining
  - Model card preview (auto-updates from benchmarks)

### Foreman Alerts
The Foreman monitors training metrics and node health:

- **Perplexity alert**: "Perplexity spiked from 3.04 to 25.78 at step 500 — likely divergence from YaRN config. Pausing to investigate."
- **Process death**: "alloy_executor.py (PID 12190) exited with OOM. VRAM was at 31.2/32GB. Recommend reducing batch size."
- **Disk alert**: "Disk at 92%. Clearing 3 old forge outputs (12GB) to make room."
- **Quality gate**: "HumanEval dropped from 74% to 41% after cycle 2. Reverting to cycle 1 checkpoint."
- **Environment**: "PyTorch 2.11+cu126 doesn't support RTX 5090 sm_120. Upgrading to cu128."

Foreman attempts fixes himself, reports what he did, escalates if he can't fix it.

### Grid Events (Cross-Node)

Training events need to flow from remote nodes to the UI:

```typescript
// Emitted by alloy_executor on the training node
Events.emit('forge:step', { nodeId, jobId, step, totalSteps, loss, perplexity, vram, sample })
Events.emit('forge:phase', { nodeId, jobId, phase, detail })
Events.emit('forge:alert', { nodeId, jobId, severity, message, metric, value, threshold })
Events.emit('forge:complete', { nodeId, jobId, results, benchmarks, modelCard })

// Emitted by Foreman
Events.emit('foreman:alert', { nodeId, severity, message, action, resolved })
Events.emit('foreman:fix', { nodeId, problem, action, result })

// Emitted by node health monitoring
Events.emit('node:health', { nodeId, gpu, disk, processes, environment })
```

These events route through the grid transport to the UI node.

## Why This Matters

- **RunPod users** need this same visibility — they're paying per hour, they need to see if training is working
- **Our grid is transactional** — forge-alloy contracts need verifiable metrics, not just "trust me it trained"
- **The mesh scales** — what works for 2 nodes works for 200, but only if the control plane is real
- **Foreman is the key** — autonomous monitoring + alerting + self-repair means the grid runs itself

## Implementation Path

1. **grid/job-log** command — stream log output from remote node (replaces `ssh + tail`)
2. **Training metric events** — alloy_executor emits structured events, not just log lines
3. **Per-node tab** — click node → open tab with terminal + widgets
4. **Training right panel** — loss curve, perplexity, samples, quality gates
5. **Foreman metric monitoring** — watches training events, triggers alerts
6. **grid/ssh** command — browser-based terminal to any grid node

## Priority
Critical — this is the difference between a demo and a product.

## Related
- #671 Factory Foreman
- #672 Factory Floor controls
- #677 Foreman as first responder
- #678 Grid collaboration layer
